src/librustc/ty/sty.rs

```diff
@@ -914,16 +944,16 @@ impl<'a, 'tcx, 'gcx> ExistentialProjection<'tcx> {
         ty::ProjectionPredicate {
             projection_ty: ty::ProjectionTy::from_ref_and_name(
                 tcx,
-                self.trait_ref.with_self_ty(tcx, self_ty),
-                self.item_name),
+                self.trait_ref(tcx).with_self_ty(tcx, self_ty),
+                self.item_name(tcx)),
             ty: self.ty,
         }
     }
 }
 
 impl<'a, 'tcx, 'gcx> PolyExistentialProjection<'tcx> {
-    pub fn item_name(&self) -> Name {
-        self.skip_binder().item_name()
+    pub fn item_name(&self, tcx: TyCtxt) -> Name {
+        self.skip_binder().item_name(tcx)
```

Comment:
Just like the `ProjectionTy` version, calls of this method should be limited to printing the name or something similar (and those don't even need there to be an `item_name` method).